### ISSUE TITLE
Does Configure::load() throw an exception and prevent loading of local file?

### ISSUE THREAD

**Issue by beporter** [beporter/CakePHP-EnvAwareness#5]
@garethellis36 / [@garethellis](https://twitter.com/garethellis) mentioned to me at CakeFest that there might be an issue with this code snippet :

``` php
try {
    $env = getenv('APP_ENV');
    Configure::load("app-{$env}", 'default'); // If this file doesn't exist (say in production)...
    Configure::load('app-local', 'default'); // ...does this still execute?
} catch (\Exception $e) {}
```

The answer is: I'm not sure-- I don't think I've tested it. If it does, it could be an issue specifically in production situations, where an `APP_ENV` var isn't meant to even be explicitly required or set. The solution is straightforward though:

``` php
try {
    $env = getenv('APP_ENV');
    Configure::load("app-{$env}", 'default');
} catch (\Exception $e) {}
try {
    Configure::load('app-local', 'default');
} catch (\Exception $e) {}
```

Not quite as succinct, but probably safer. Has anyone tested this? It shouldn't be too difficult using the demo vagrant box in this repo.

**Comment by cwbit** [comment]
Yeah, I end up always doing the second with the two try-catch statements because sometimes (esp on CLI) the ENV isn't set and it causes the whole try/catch to fail, meaning my `app-local` never gets loaded.

I actually error if the `app-local` file isn't there because its job is to hold the actual database passwords and other platform-specific settings that can't be in source-control, are critical, and are situation specific. I try and keep as much as possible out of these files but stuff like username/password for database always goes inside my `app-local`

Here's a snippet from an actual app of mine

``` php
/**
 * Load app-env (OPTIONAL)
 *
 * Contains settings that should be overridden at the environment level
 * e.g. setting debug => true for the 'dev' environment but not 'uat'
 *
 * make sure to set in VirtualHost ... SetEnv APP_ENV foo
 */
try {
    $env = getenv('APP_ENV');
    Configure::load("app-{$env}", 'default');
} catch (\Exception $e) {
    # ignore, if not used
}

/**
 * Load app-local (REQUIRED)
 *
 * It is expected that this file will contain, at the min, the database username/password
 */
try {
    Configure::load("app-local", 'default');
} catch (\Exception $e) {
    die($e->getMessage() . "\n");
}
```

**Comment by beporter** [comment]
Yep, easily adaptable to each project's needs. :smile: :+1:

If I were working with a team, I would probably supplement the `die()` message to make it more explicit why things failed:

``` php
// formatted for readability here
die(
  'Untracked (but mandatory) config/app-local.php missing. '
  . ' Copy & review config/app-local-sample.php to start. Exception message: '
  . $e->getMessage() . "\n"
);
```

A new dev would only stumble on that exactly once, and also immediately understand how the system worked without any further documentation or "study". Quick recovery plus an environment lesson, in 3 lines of code.

**Comment by cwbit** [comment]
Great idea with smart die() :+1:
